src/main/java/edu/wpi/grip/core/Socket.java

```diff
@@ -103,6 +110,13 @@ public Step getStep() {
     }
 
     /**
+     * @return The set of connections that have this socket as an input or output
+     */
+    public Set<Connection> getConnections() {
+        return this.connections;
```

Comment:
Guava has an immutable set. Manipulations of this set should only happen inside of this class.